File: Level2/Materal.BusinessFlow/Src/Materal.BusinessFlow.Abstractions/Domain/FlowRecord.cs
using Materal.BusinessFlow.Abstractions.Enums;
using System.ComponentModel.DataAnnotations;

namespace Materal.BusinessFlow.Abstractions.Domain
{
    public class FlowRecord : BaseDomain
    {
        /// <summary>
        /// 流程唯一标识
        /// </summary>
        [Required]
        public Guid FlowID { get; set; }
        /// <summary>
        /// 步骤唯一标识
        /// </summary>
        [Required]
        public Guid StepID { get; set; }
        /// <summary>
        /// 节点唯一标识
        /// </summary>
        [Required]
        public Guid NodeID { get; set; }
        /// <summary>
        /// 可操作用户
        /// </summary>
        public Guid? UserID { get; set; }
        /// <summary>
        /// 操作用户
        /// </summary>
        public Guid? OperationUserID { get; set; }
        /// <summary>
        /// 状态
        /// </summary>
        [Required]
        public FlowRecordStateEnum State { get; set; } = FlowRecordStateEnum.Wait;
        /// <summary>
        /// 位序
        /// </summary>
        [Required]
        public int SortIndex { get; set; }
        /// <summary>
        /// 节点处理类型
        /// </summary>
        [Required]
        public NodeHandleTypeEnum NodeHandleType { get; set; }
        /// <summary>
        /// 数据
        /// </summary>
        public string? Data { get; set; }
        /// <summary>
        /// 返回消息
        /// </summary>
        public string? ResultMessage { get; set; }
    }
}

File: Level3/Materal.BusinessFlow/Demo/BusinessFlowWebDemo/src/models/Flow/FlowRecord.ts
import { BaseDomain } from "../BaseDomain";
import { NodeHandleTypeEnum } from "../Node/NodeHandleTypeEnum";
import { FlowRecordStateEnum } from "./FlowRecordStateEnum";

export class FlowRecord extends BaseDomain {
    /**
     * 流程唯一标识
     */
    public FlowID: string = "";
    /**
     * 步骤唯一标识
     */
    public StepID: string = "";
    /**
     * 节点唯一标识
     */
    public NodeID: string = "";
    /**
     * 可操作用户
     */
    public UserID?: string;
    /**
     * 操作用户
     */
    public OperationUserID?: string;
    /**
     * 状态
     */
    public State: FlowRecordStateEnum = FlowRecordStateEnum.Wait;
    /**
     * 位序
     */
    public SortIndex: number = 0;
    /**
     * 节点处理类型
     */
    public NodeHandleType: NodeHandleTypeEnum = NodeHandleTypeEnum.Auto;
    /**
     * 数据
     */
    public Data?: string;
    /**
     * 返回消息
     */
    public ResultMessage?: string;
    /**
     * 流程模版唯一标识
     */
    public FlowTemplateID: string = "";
    /**
     * 流程模版名称
     */
    public FlowTemplateName: string = "";
    /**
     * 步骤名称
     */
    public StepName: string = "";
    /**
     * 节点名称
     */
    public NodeName: string = "";
    /**
     * 状态文本
     */
    public StateText: string = "";
}